op-e2e/actions/l2_engine.go

```diff
@@ -168,6 +171,13 @@ func (e *L2Engine) ActL2RPCFail(t Testing) {
 	e.failL2RPC = errors.New("mock L2 RPC error")
 }
 
+// ActSendTx adds a transaction to the tx pool and ensures any required promotion to the pending tx queue is complete
+// before returning.
+func (e *L2Engine) ActSendTx(t Testing, tx *types.Transaction) {
+	err := errors.Join(e.eth.TxPool().Add([]*types.Transaction{tx}, true, true)...)
+	require.NoError(t, err, "Add tx to pool")
+}
```

Comment:
Given we have the waiter in place I suspect we shouldn't do this now and should just stick to using the actual JSON RPC api.